tests/unit/iontests.js

```diff
@@ -11,126 +11,376 @@
+            var shouldSkip = false;
+            for (var skip of skipList) {
+                if (spath.endsWith(skip)) {
+                    shouldSkip = true;
+                    break;
+                }
+            }
+            if (debugList.length > 0) {
+                shouldSkip = true;
+                for (var debug of debugList) {
+                    if (spath.endsWith(debug)) {
+                        shouldSkip = false;
+                        break;
+                    }
+                }
+            }
+            if (!shouldSkip) {
+                badUnskipped.push(path);
+            }
         }
-      }
-      if (!shouldSkip) {
-        unskipped.push(path);
-      }
-    }
 
-    var suite = {
-      name: 'Good tests',
-    };
-
-    function exhaust(reader) {
-      var tries = 0;
-      for (;;) {
-        // Safety valve
-        tries++;
-        if (tries > 1000) {
-          throw new Error("Reader spinning forever!");
+        var goodSuite = {
+            name: 'Good tests'
+        };
+        var badSuite = {
+            name: 'Bad tests'
+        };
+
+        var eventStreamSuite = {
+            name: 'EventStream tests'
+        };
+        var badEventStreamSuite = {
+            name: 'Bad EventStream tests'
+        };
+
+        function goodExhaust(reader) {
+            var tries = 0;
+            for (;;) {
+                // Safety valve
+                tries++;
+                if (tries > 1000) {
+                    throw new Error("Reader spinning forever!");
+                }
+
+                var next = reader.next();
+                if (typeof(next) === 'undefined') {
+                    if (reader.depth() > 0) {
+                        // End of container
+                        //console.log("Stepping out");
+                        reader.stepOut();
+                    } else {
+                        // End of data
+                        break;
+                    }
+                } else if (next.container && !reader.isNull()) {
+                    //console.log("Stepping in");
+                    reader.stepIn();
+                }
+            }
         }
 
-        var next = reader.next();
-        if (typeof(next) === 'undefined') {
-          if (reader.depth() > 0) {
-            // End of container
-            //console.log("Stepping out");
-            reader.stepOut();
-          } else {
-            // End of data
-            break;
-          }
-        } else if (next.container && !reader.isNull()) {
-          //console.log("Stepping in");
-          reader.stepIn();
+        function badExhaust(reader) {
+            var tries = 0;
+            try {
+                for (; ;) {
+                    // Safety valve
+                    tries++;
+                    if (tries > 1000) {
+                        throw new Error("Reader spinning forever!");
+                    }
+
+                    var next = reader.next();
```

Comment:
Should this be a `let`? `var` is going to hoist this to the top of the function.